internal/datadog/metrics_client_test.go

```diff
@@ -72,6 +74,44 @@ func TestGaugeMultiple(t *testing.T) {
 	metricdatatest.AssertEqual(t, want, got, metricdatatest.IgnoreTimestamp())
 }
 
+func TestGaugeDataRace(t *testing.T) {
+	reader, metricClient, _ := setupMetricClient()
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	go func() {
+		defer wg.Done()
+		cnt := 0
```

Comment:
What is the purpose of `cnt`? Is it supposed to be incremented in the for loop or remain as a constant?